lib/Target/TTNN/TTNNToFlatbuffer.cpp

```diff
@@ -362,6 +362,32 @@ createOp(FlatbufferObjectCache &cache, ArangeOp op) {
       memoryConfigDesc /* optional */, output);
 }
 
+::flatbuffers::Offset<::tt::target::ttnn::OnesOp>
+createOp(FlatbufferObjectCache &cache, OnesOp op) {
+  ::flatbuffers::Offset<::flatbuffers::Vector<int64_t>> shape =
+      cache.fbb->CreateVector<int64_t>(op.getShape().getShape());
+
+  ::flatbuffers::Optional<::tt::target::DataType> dtype =
+      toFlatbufferOptional(cache, op.getDtype());
+
+  ::flatbuffers::Optional<::tt::target::TensorLayout> layout =
+      toFlatbufferOptional(cache, op.getLayout());
+
+  flatbuffers::Offset<::tt::target::DeviceRef> device =
```

Comment:
How does serialization work in case there is no device? I see you added 0 as a fallback here. Can we also use flatbuffer::nullopt like you did for layout?

Reply:
The type here is `flatbuffers::Offset` which means that the vtable entry of device would be 0. In Flatbuffer speak, that means that the value isn't set. Enums were trickier, I can't remember the details, but I think that when I tried without setting them to `null` in `program.fbs`, they would be treated such that `0` would mean the first value of the enum instead of unset (or offset=0, as is for non-enum types). Setting the enums to `null` is what provides the Optional wrapper.